wsd/DocumentBroker.hpp

```diff
@@ -540,6 +540,46 @@ class DocumentBroker : public std::enable_shared_from_this<DocumentBroker>
     /// a convert-to request or doctored to look like one.
     virtual bool isConvertTo() const { return false; }
 
+private:
+    /// Responsible for managing document saving.
+    /// Tracks auto-saveing and its frequency.
+    class SaveManager final
+    {
+    public:
+        SaveManager()
+            : _lastAutosaveCheckTime(std::chrono::steady_clock::now())
+            , _isAutosaveEnabled(std::getenv("LOOL_NO_AUTOSAVE") == nullptr)
+        {
+        }
+
+        /// Return true iff auto save is enabled.
+        bool isAutosaveEnabled() const { return _isAutosaveEnabled; }
+
+        /// Returns true if we should issue an auto-save.
+        bool needAutosaveCheck() const
```

Comment:
Maybe shouldAutosave() or shouldPerformAutosave() would be a bit better name?

Reply:
Actually, this is only to *check* if we need to autosave. It doesn't decide to autosave, it only decides if we should check the conditions to autosave or not.

But I never liked the name either! I'll rework it when I centralize the save/upload decision logic.